track.py

```diff
@@ -61,12 +62,38 @@ class Track:
     @staticmethod
     def from_xml_file(filename):
         xmldoc = parse(filename)
+        width = get_track_root_info(xmldoc, 'width')
+        profil_steps_length = get_track_root_info(xmldoc, 'profil steps length')
+
         segments = []
         for section in get_main_track(xmldoc).childNodes:
             if section.nodeType == Node.ELEMENT_NODE:
-                segments.append(Segment.from_xml(section))
+                s = Segment.from_xml(section)
+                if s.end_radius is None:
+                    segments.append(s)
+                else:
+                    length = abs((s.radius + s.end_radius)/2.0 * s.arc)
+                    num_steps = int(length/4.0) + 1
+                    if num_steps == 1:
+                        segments.append(s)
+                    else:
+                        dlen = length/float(num_steps - 1)
```

Comment:
Let's say I end up with `num_steps == 2`. Probably the step length is set to 10, length is 15, so I need to split the segment into two pieces. I want `dlen == length/2 == 7.5` , right? Why would I want to divide by `num_steps-1`?